pkg/generate/app/cmd/newapp.go

```diff
@@ -243,6 +243,34 @@ func (c *AppConfig) AddArguments(args []string) []string {
 	return unknown
 }
 
+// validateBuilders confirms that all images associated with components that are to be built,
+// are builders (or we're using a docker strategy).
+func (c *AppConfig) validateBuilders(components app.ComponentReferences) error {
+	errs := []error{}
+	for _, ref := range components {
+		input := ref.Input()
+
+		// if the strategy is explicitly Docker, all repos should assume docker
+		if c.Strategy == "docker" && input.Uses != nil {
+			input.Uses.BuildWithDocker()
+		}
+
+		// if we are expecting build inputs, or get a build input when strategy is not docker, expect to build
+		if c.ExpectToBuild || (input.ResolvedMatch.Builder && c.Strategy != "docker") {
+			input.ExpectToBuild = true
+		}
+
+		switch {
+		case input.ExpectToBuild && input.ResolvedMatch != nil && !input.ResolvedMatch.Builder && input.Uses != nil && !input.Uses.IsDockerBuild():
+			if len(c.Strategy) == 0 {
```

Comment:
yeah, i think i had a greater vision once upon a time for what this would check :)